tests/Moq.Tests/CallbacksFixture.cs

```diff
@@ -474,6 +474,105 @@ public void CallbackWithRefParameterCannotModifyNonRefParameter()
 			Assert.Equal("input", value);
 		}
 
+		[Fact]
+		public void CallbackWithIndexerGetter()
+		{
+			int x = default(int);
+
+			var mock = new Mock<IFoo>();
+			mock.Setup(f => f[It.IsAny<int>()])
+				.Callback(new Action<int>(x_ => x = x_))
+				.Returns(32);
+
+			int result = mock.Object[17];
+			Assert.Equal(17, x);
+			Assert.Equal(32, result);
+		}
+
+		[Fact]
+		public void CallbackWithIndexerSetter()
+		{
+			int x = default(int);
+			int result = default(int);
+
+			var mock = new Mock<IFoo>();
+			mock.SetupSet(f => f[10] = It.IsAny<int>())
+				.Callback(new Action<int, int>((x_, result_) =>
+				{
+					x = x_;
+					result = result_;
+				}));
+
+			mock.Object[10] = 5;
+			Assert.Equal(10, x);
+			Assert.Equal(5, result);
+		}
+
+		[Fact]
+		public void CallbackWithMultipleArgumentIndexerGetter()
+		{
+			int x = default(int);
+			int y = default(int);
+
+			var mock = new Mock<IFoo>();
+			mock.Setup(f => f[It.IsAny<int>(), It.IsAny<int>()])
```

Comment:
Same as in an earlier review comment, I appreciate the test, but it doesn't seem to be related to this PR, since it uses the expression tree-based `Setup` instead of the delegate-based `SetupSet`...?

Could you state how it relates to the bugfix in your PR (and if it doesn't, could you remove it and optionally re-submit in a separate PR?)